libs/wingc/src/jsify.rs

```diff
@@ -1486,22 +1489,31 @@ impl<'ast> Visit<'ast> for FieldReferenceVisitor<'ast> {
 			key = format!("this.{}", key);
 		}
 
-		// if our last captured component is a resource and we don't have
-		// a qualification for it, it's currently an error.
 		if let Some(c) = capture.last() {
 			if let Some(v) = &c.variable {
-				if v.type_.as_resource().is_some() {
-					if qualification.len() == 0 {
-						self.diagnostics.push(Diagnostic {
-							message: format!(
-								"Unable to qualify which operations are performed on '{}' of type '{}'. This is not supported yet.",
-								key, v.type_,
-							),
-							span: Some(node.span.clone()),
-						});
-
-						return;
-					}
+				// if our last captured component is a non-handler resource and we don't have
+				// a qualification for it, it's currently an error.
+				if v.type_.is_resource() && !v.type_.is_handler_resource() && qualification.len() == 0 {
+					self.diagnostics.push(Diagnostic {
+						message: format!(
+							"Unable to qualify which operations are performed on '{}' of type '{}'. This is not supported yet.",
+							key, v.type_,
+						),
+						span: Some(node.span.clone()),
+					});
+
+					return;
+				}
+
+				// if this reference refers to an inflight function or handler resource,
+				// we need to give permission to the "handle" operation
+				if v.type_.is_inflight_function() || v.type_.is_handler_resource() {
```

Comment:
I think I understand why we need to add the "handle" operation in casse it's a handler resource. But if it's a plain old inflight function, then why add this permission?

Reply:
We need to add the "handle" operation in both situations in case a handler resource is assigned to a field of an inflight closure type. For example, in this snippet:

```js
class MyResource {
  closure: inflight (str): str;

  init() {
    this.closure = inflight (s: str): str => {
      globalBucket.list();
      return "hello";
    };
  }

  inflight foo(): str {
    return this.closure("anything");
  }
}
```

If I change this line in the `jsify.rs` so we only add the "handle" operation if it's a handler resource, then the compiler no longer annotates `foo` with the "handle" permission, causing the test to fail.